processor/groupbyattrsprocessor/processor_test.go

```diff
@@ -691,3 +754,83 @@ func retrieveMetric(metrics pdata.MetricSlice, name string, metricType pdata.Met
 	}
 	return pdata.Metric{}, false
 }
+
+func TestCompacting(t *testing.T) {
+	spans := someSpans(attrMap, 10, 10)
+	logs := someLogs(attrMap, 10, 10)
+	metrics := someGaugeMetrics(attrMap, 10, 10)
+
+	assert.Equal(t, 100, spans.ResourceSpans().Len())
+	assert.Equal(t, 100, logs.ResourceLogs().Len())
+	assert.Equal(t, 100, metrics.ResourceMetrics().Len())
+
+	gap := createGroupByAttrsProcessor(zap.NewNop(), []string{})
+
+	processedSpans, err := gap.processTraces(context.Background(), spans)
+	assert.NoError(t, err)
+	processedLogs, err := gap.processLogs(context.Background(), logs)
+	assert.NoError(t, err)
+	processedMetrics, err := gap.processMetrics(context.Background(), metrics)
+	assert.NoError(t, err)
+
+	assert.Equal(t, 1, processedSpans.ResourceSpans().Len())
+	assert.Equal(t, 1, processedLogs.ResourceLogs().Len())
+	assert.Equal(t, 1, processedMetrics.ResourceMetrics().Len())
+
+	rss := processedSpans.ResourceSpans().At(0)
+	rls := processedLogs.ResourceLogs().At(0)
+	rlm := processedMetrics.ResourceMetrics().At(0)
+
+	assert.Equal(t, 0, rss.Resource().Attributes().Len())
+	assert.Equal(t, 0, rls.Resource().Attributes().Len())
+	assert.Equal(t, 0, rlm.Resource().Attributes().Len())
+
+	assert.Equal(t, 10, rss.InstrumentationLibrarySpans().Len())
+	assert.Equal(t, 10, rls.InstrumentationLibraryLogs().Len())
+	assert.Equal(t, 10, rlm.InstrumentationLibraryMetrics().Len())
+
+	for i := 0; i < 10; i++ {
+		ils := rss.InstrumentationLibrarySpans().At(i)
+		ill := rls.InstrumentationLibraryLogs().At(i)
+		ilm := rlm.InstrumentationLibraryMetrics().At(i)
+
+		assert.Equal(t, 10, ils.Spans().Len())
+		assert.Equal(t, 10, ill.LogRecords().Len())
+		assert.Equal(t, 10, ilm.Metrics().Len())
+	}
+}
+
+func BenchmarkCompacting(bb *testing.B) {
```

Comment:
Can you add your results as a comment to this issue?

Reply:
Sure thing! Added